src/App/Expression.cpp

```diff
@@ -1998,6 +2031,76 @@ Py::Object FunctionExpression::evalAggregate(
     return pyFromQuantity(c->getQuantity());
 }
 
+Base::Vector3d FunctionExpression::evaluateSecondVectorArgument(const Expression *expression, const std::vector<Expression*> &arguments)
+{
+    Py::Tuple vectorValues;
+    Py::Object secondParameter = arguments[1]->getPyValue();
+
+    if (arguments.size() == 2) {
+        if (!secondParameter.isSequence())
+            _EXPR_THROW("Second parameter is not a sequence type: '" << secondParameter.as_string() << "'.", expression);
+        if (PySequence_Size(secondParameter.ptr()) != 3)
+            _EXPR_THROW("Second parameter provided has " << PySequence_Size(secondParameter.ptr()) << " elements instead of 3.", expression);
+            
+        vectorValues = Py::Tuple(Py::Sequence(secondParameter));
+    } else {
```

Comment:
What about `args.size() == 1`? The calling code only checks to make sure `args` is not empty.

Reply:
This method is only used when a second (or more) argument(s) is/are present hence the name of the method `evaluateSecondVectorArgument`. Should I add a guard which throws an exception when `args.size() == 1` at the top of this method?

Reply:
Ah sorry I meant to say, although you're right that the calling code only checks if `args` is not empty, the constructor also checks the number of parameters.

Reply:
Not a hill I'll die on, but as a general rule I like error-checking code to be obviously complete so I don't have to spend the time tracing back to make sure that the condition that's not being checked is somehow being avoided in another way.